### PR TITLE
Update PHPUnit to 5.7

### PR DESCRIPTION
**Pull request by davidyell**
Cake 3 is more compatible with later versions now, and it's required for the newer cake test suite.

This works for me locally with CakePHP 3.4.6, PHP Unit 5.7.19 and PHP 7.1.4

Will hopefully get the PR's CI to pass as well with Travis. I picked PHP Unit 5.7.19, as it's the most up to date version which still supports PHP 5.6, to give some flexibility.

### CODE REVIEW

**Comment by davidyell** [comment]
Right, well this might take someone with a better understanding of Travis than I do, otherwise I'm going to end up just filling this PR with commits trying to make it work.

I have no idea how it's getting phpunit 5.3.4!

**Comment by davidyell** [comment]
Potential fix over here, https://github.com/jeremyharris/cakephp-lazyload/pull/14

**Comment by davidyell** [comment]
I'll take a proper look at this today and then rebase the commits and push it back to the branch.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 language: php
 
 php:
+  - 7.1
   - 7.0
-  - 5.5
```
Comment (ADmad) [inline .travis.yml]: The plugin is compatible with older cake versions which support php 5.5 so we should keeping testing on 5.5.

File: composer.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "require-dev": {
         "cakephp/cakephp": ">=3.2",
         "cakephp/cakephp-codesniffer": "2.*",
-        "phpunit/phpunit": "4.1.*"
+        "phpunit/phpunit": "5.7.*"
```
Comment (ADmad) [inline composer.json]: Changing this to `"<6.0"` is better, that way composer can pick appropriate version based on php version. Using `"<6.0"` has worked fine on other plugins i maintain.

**Pull request merged**